src/test/scala/org/apache/spark/sql/kinesis/KinesisSourceSuite.scala

```diff
@@ -87,6 +87,43 @@ class KinesisSourceOptionsSuite extends StreamTest with SharedSQLContext {
     testBadOptions()("Stream name is a required field")
     testBadOptions("streamname" -> "")("Stream name is a required field")
   }
+
+  test("Kinesis Source Options should always be case insensitive") {
+
+    def testKinesisOptions(newOptions: (String, String)*): Unit = {
+      var options = Map.empty[String, String]
+      options = options + ("streamname" -> "tmpStream")
+      newOptions.foreach {
+        case (k, v) => options = options + (k -> v)
+      }
+      // caseInsensitiveOptions are used to create a source
+      val caseInsensitiveOptions = options.map(
+        kv => kv.copy(_1 = kv._1.toLowerCase(Locale.ROOT))
+      )
+      val source = (new KinesisSourceProvider)
+        .createSource(
+          spark.sqlContext,
+          "/tmp/path",
+          None,
+          "kinesis",
+          caseInsensitiveOptions
+        )
+      val kinesisSourceOptions = source.asInstanceOf[KinesisSource].options
+      newOptions.foreach {
+        case (k, v) =>
+          val keytoCheck = k.toLowerCase(Locale.ROOT).drop(8).toString
+          assert(kinesisSourceOptions.getOrElse(keytoCheck, "None").toString == v.toString)
```

Comment:
Should not we use .equals instead of == for String ?
```suggestion
          assert( (kinesisSourceOptions.getOrElse(keytoCheck, "None").toString).equals(v.toString) )
```